### ISSUE TITLE
Better message when the core:archive failed with an internal server error

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#6723]
The goal of this issue is to improve the message displayed to users when an API requests fails during the `core:archive` command. 

Users often have this problem and need to ask in the forums. For example today [this post](http://forum.piwik.org/read.php?2,122191). When the `core:archive` command fails, the output looks like this:

> Got invalid response from API request: http://piwikdomain.com/index.php?module=API&method=X&format=original&serialize=1&trigger=archivephp. The response was empty. This usually means a server error. This solution to this error is generally to increase the value of 'memory_limit' in your php.ini file. Please check your Web server Error Log file for more details. 

The message mentions the `web server error log file` but often the `core:archive` command will use CLI to archive the data (in  #4610). So the error will rather be logged in the PHP cli error log file.

Maybe we could make the message dependant on the context Cli VS Http and point to the actual error log file path if we can detect it.

**Comment by mnapoli** [comment]
OK got it, but we can't know the location of the server error file. It's not configured in Piwik, it's something like `/var/log/apache/error.log` but it can be anywhere.

**Comment by mnapoli** [comment]
I'm thinking about it over and over...

- if running through web
    - exception/error: the HTTP request will return an invalid response containing the error
        - the error message shown will contain the error description: **Good**
    - memory limit or other fatal error with empty response
        - the error message shown will say "check your Web server Error Log file": **Good**
- if running through `climulti:request`
    - exception/error: the CLI command will return an invalid response containing the error
        - the error message shown will contain the error description: **Good**
    - memory limit or other fatal error with empty response
        - PHP running `climulti:request` **should display on the CLI output the PHP error message** (maybe on stderr), which means CLIMulti *should* pick it up and never go in the `if (empty($response))` (because `$response` would contain the PHP error): **Good**

        For example running this shows the PHP error directly:

        ```
        $ php -n -dmemory_limit=1M -r 'while (true) { $str .= 'h'; }'

        Fatal error: Allowed memory size of 1048576 bytes exhausted (tried to allocate 786385 bytes) in Command line code on line 1
        ```

TL/DR: empty response should only happen when running through web (i.e. `climulti:request` not supported).

```php
$message = "Got invalid response from API request: $url. ";
if (empty($response)) {
    $message .= "The response was empty. This usually means a server error. This solution to this error is generally to increase the value of 'memory_limit' in your php.ini file. Please check your Web server Error Log file for more details.";
} else {
    $message .= "Response was '$response'";
}
```

This error message should be fine then.

**Comment by mattab** [comment]
Sounds good! +1

**Comment by mnapoli** [comment]
OK all should be good then
